README.md
JHijri Picker Package
-
Just An Amazing Hijri Date Picker Widget With Many Customizable Options, Easy To Use, Customise & Implement.


## Features
* Hijri Date Picker.
* Normal Date Picker.
* Easy To Use , Customise and Implement.


## Usage
To Use `JGlobalDatePicker`

* JDateModel

This Model Could Be JHijri Or DateTime, One Value Not Both.
```dart
JDateModel dM = JDateModel(dateTime: DateTime.now());
Or
dM = JDateModel(jhijri: JHijri.now());
```
If You Want to Use JHijri IN JDateModel You Have to add [JHijriConverter](https://pub.dev/packages/jhijri) to your `pubspec.yaml`

Note:
-
If You Use `onChange` Method You Can't Use `onOk`

As Simple Dialog:
-
```dart
final val = openDialog(context);
Future<HijriDate?> openDialog(BuildContext context)async{
 return await showGlobalDatePicker(context: context, pickerType:PickerType.JNormal);
}
```
As Full Dialog:
-
```dart
final dialog = openDialog(context);
Future<HijriDate?> openDialog(BuildContext context)async{
  return await showGlobalDatePicker(
    context: context,
    startDate: JDateModel(
        jhijri: JHijri(
          fYear: 1442,
          fMonth: 12,
          fDay: 10,
        )),
    selectedDate: JDateModel(jhijri: JHijri.now()),
    endDate: JDateModel(
        jhijri: JHijri(
          fDay: 25,
          fMonth: 1,
          fYear: 1460,
        )),
    pickerMode: DatePickerMode.day,
    pickerTheme: Theme.of(context),
    textDirection: TextDirection.rtl,
    //locale: const Locale("ar", "SA"),
    okButtonText: "حفظ",
    cancelButtonText: "عودة",
    onOk: (value) {
      debugPrint(value.toString());
      Navigator.pop(context);
    },
    onCancel: () {
      Navigator.pop(context);
    },
    primaryColor: Colors.blue,
    calendarTextColor: Colors.white,
    backgroundColor: Colors.black,
    borderRadius: const Radius.circular(10),
    buttonTextColor: Colors.white,
    headerTitle: const Center(
      child: Text("التقويم الهجري", style: TextStyle(color: Colors.white),),
    ),
  );
}
```

As Widget:
-
```dart
Widget _JHijriAsWidget(){
  return JGlobalDatePicker(
    widgetType: WidgetType.JContainer,
    pickerType: PickerType.JHijri,
    buttons: const SizedBox(),
    primaryColor: Colors.blue,
    calendarTextColor: Colors.white,
    backgroundColor: Colors.black,
    borderRadius: const Radius.circular(10),
    headerTitle: const Center(
      child: Text("التقويم الهجري"),
    ),
    startDate: JDateModel(dateTime: DateTime.parse("1984-12-24")),
    selectedDate: JDateModel(dateTime: DateTime.now()),
    endDate: JDateModel(dateTime: DateTime.parse("2030-09-20")),
    pickerMode: DatePickerMode.day,
    pickerTheme: Theme.of(context),
    textDirection: TextDirection.rtl,
    onChange: (val) {
      debugPrint(val.toString());
    },
  );
}
```

## Additional information

Provided By [Just Codes Developers](https://jucodes.com/)


CHANGELOG.md
## 1.0.1
* Update Dependencies
## 1.0.0
* Merge `JHijriPicker` and `JWesternPicker` on `JGlobalPicker` 
* Change `showJHijriPicker`  to `showGlobalDatePicker` 
* Fixing analyse Issue
* Adding `JPickerValue` class as return type for `showGlobalDatePicker`
* Adding `JPickerValue` class as return type for `onOk`
* Adding `JPickerValue` class as return type for `onChange`
## 0.0.4
* JHijriPicker Fourth Release
* Merge `showJDateTimePicker` with `showJHijriPicker`
## 0.0.3
* JHijriPicker Third Release
* Remove `requird WidgetType`
## 0.0.2
* JHijriPicker Second Release
* Adding `JDateTimePicker`
* Fixing `example.dart`
## 0.0.1

* JHijriPicker First Release


lib/_src/_jWidgets.dart
import 'package:flutter/material.dart';
import 'package:jhijri/jHijri.dart';
import '_src.dart';

enum WidgetType {
  JDialog,
  JContainer,
}

enum PickerType {
  JHijri,
  JNormal,
}

class JDateModel {
  JHijri? jhijri;
  DateTime? dateTime;
  JDateModel({this.dateTime, this.jhijri});
}

class JPickerValue {
  JHijri? _jHijri;
  DateTime? _dateTime;
  JHijri get jhijri => _jHijri ?? JHijri.now();
  DateTime get date => _dateTime ?? DateTime.now();
  JPickerValue.value(JDateModel data) {
    if (data.jhijri != null) {
      _jHijri = data.jhijri!;
      _dateTime = data.jhijri!.dateTime;
    }
    if (data.dateTime != null) {
      _dateTime = data.dateTime;
      if (data.dateTime!.year > 1938 && data.dateTime!.year < 2076) {
        _jHijri = JHijri(fDate: data.dateTime);
      }
    }
  }
  @override
  String toString() {
    return {
      "Date": date.toString(),
      "HijriDate": jhijri.toMap(),
    }.toString();
  }
}

Future<JPickerValue?> showGlobalDatePicker({
  required BuildContext context,
  PickerType pickerType = PickerType.JHijri,
  JDateModel? startDate,
  JDateModel? endDate,
  JDateModel? selectedDate,
  DatePickerMode? pickerMode,
  String? okButtonText,
  String? cancelButtonText,
  Widget? headerTitle,
  Function(JPickerValue datetime)? onOk,
  Function(JPickerValue datetime)? onChange,
  VoidCallback? onCancel,
  ThemeData? pickerTheme,
  Color? primaryColor,
  Color? backgroundColor,
  Color? calendarTextColor,
  Color? buttonTextColor,
  Color? cancelButtonColor,
  Color? okButtonColor,
  Radius? borderRadius,
  Widget? buttons,
  TextDirection? textDirection,
  Locale? locale,
  SelectableDayPredicate? selectableDayPredicate,
  JSelectableDayPredicate? jSelectableDayPredicate,
}) {
  JHijri _select = JHijri();
  JHijri _start = JHijri(fYear: 2000);
  JHijri _end = JHijri(fYear: 2060);
  JDateModel _startDate = startDate != null
      ? startDate
      : pickerType == PickerType.JHijri
          ? JDateModel(jhijri: _start)
          : JDateModel(dateTime: _start.dateTime);
  JDateModel _endDate = endDate != null
      ? endDate
      : pickerType == PickerType.JHijri
          ? JDateModel(jhijri: _end)
          : JDateModel(dateTime: _end.dateTime);
  JDateModel _selectedDate = selectedDate != null
      ? selectedDate
      : pickerType == PickerType.JHijri
          ? JDateModel(jhijri: _select)
          : JDateModel(dateTime: _select.dateTime);
  return showDialog<JPickerValue?>(
    context: context,
    builder: (BuildContext context) {
      return JGlobalDatePicker(
          pickerType: pickerType,
          selectableDayPredicate: selectableDayPredicate,
          jSelectableDayPredicate: jSelectableDayPredicate,
          locale: locale,
          textDirection: textDirection,
          buttons: buttons,
          borderRadius: borderRadius,
          okButtonText: okButtonText,
          okButtonColor: okButtonColor,
          cancelButtonText: cancelButtonText,
          cancelButtonColor: cancelButtonColor,
          calendarTextColor: calendarTextColor,
          primaryColor: primaryColor,
          pickerTheme: pickerTheme,
          pickerMode: pickerMode,
          onCancel: onCancel,
          onChange: onChange,
          onOk: onOk,
          headerTitle: headerTitle,
          backgroundColor: backgroundColor,
          buttonTextColor: buttonTextColor,
          selectedDate: _selectedDate,
          startDate: _startDate,
          endDate: _endDate);
    },
  );
}

class _JLocalDirection extends StatelessWidget {
  final TextDirection? textDirection;
  final Locale? locale;
  final Widget child;
  const _JLocalDirection(
      {Key? key, this.textDirection, this.locale, required this.child})
      : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Localizations.override(
      context: context,
      locale: locale,
      child: textDirection != null
          ? Directionality(
              textDirection: textDirection!,
              child: child,
            )
          : child,
    );
  }
}

class _WidgetType extends StatelessWidget {
  final Widget child;
  final WidgetType widgetType;
  const _WidgetType({Key? key, required this.widgetType, required this.child})
      : super(key: key);

  @override
  Widget build(BuildContext context) {
    switch (widgetType) {
      case WidgetType.JDialog:
        return Dialog(
          backgroundColor: Colors.transparent,
          alignment: Alignment.center,
          child: child,
        );
      case WidgetType.JContainer:
        return Container(
          color: Colors.transparent,
          alignment: Alignment.center,
          child: child,
        );
    }
  }
}

class JGlobalDatePicker extends StatelessWidget {
  final PickerType pickerType;
  final JDateModel? startDate;
  final JDateModel? endDate;
  final JDateModel? selectedDate;
  final DatePickerMode? pickerMode;
  final String? okButtonText;
  final String? cancelButtonText;
  final Widget? headerTitle;
  final Function(JPickerValue datetime)? onOk;
  final Function(JPickerValue datetime)? onChange;
  final VoidCallback? onCancel;
  final ThemeData? pickerTheme;
  final Color? primaryColor;
  final Color? backgroundColor;
  final Color? calendarTextColor;
  final Color? buttonTextColor;
  final Color? cancelButtonColor;
  final Color? okButtonColor;
  final Radius? borderRadius;
  final Widget? buttons;
  final WidgetType widgetType;
  final TextDirection? textDirection;
  final Locale? locale;
  final SelectableDayPredicate? selectableDayPredicate;
  final JSelectableDayPredicate? jSelectableDayPredicate;
  const JGlobalDatePicker(
      {Key? key,
      this.pickerType = PickerType.JHijri,
      this.startDate,
      this.endDate,
      this.selectedDate,
      this.pickerMode,
      this.headerTitle,
      this.onOk,
      this.onChange,
      this.onCancel,
      this.primaryColor,
      this.backgroundColor,
      this.calendarTextColor,
      this.buttonTextColor,
      this.borderRadius,
      this.buttons,
      this.widgetType = WidgetType.JDialog,
      this.textDirection,
      this.locale,
      this.cancelButtonColor,
      this.okButtonColor,
      this.selectableDayPredicate,
      this.jSelectableDayPredicate,
      this.okButtonText,
      this.cancelButtonText,
      this.pickerTheme})
      : super(key: key);

  @override
  Widget build(BuildContext context) {
    final theme = pickerTheme ?? Theme.of(context);
    final media = MediaQuery.of(context);
    JPickerValue? selected;
    return _JLocalDirection(
        locale: locale,
        textDirection: textDirection,
        child: _WidgetType(
            widgetType: widgetType,
            child: Theme(
              data: ThemeData(
                colorScheme: theme.colorScheme.copyWith(
                  primary: primaryColor ?? Colors.blue,
                  surface: backgroundColor ?? Colors.white,
                  onSurface: calendarTextColor ?? Colors.black,
                ),
              ),
              child: SingleChildScrollView(
                child: Column(
                  mainAxisSize: MainAxisSize.min,
                  children: [
                    if (headerTitle != null) headerTitle!,
                    Container(
                      constraints:
                          BoxConstraints(maxHeight: media.size.height - 120),
                      decoration: BoxDecoration(
                        color: backgroundColor ?? Colors.white,
                        borderRadius: BorderRadius.only(
                            topLeft: borderRadius ?? const Radius.circular(16),
                            topRight:
                                borderRadius ?? const Radius.circular(16)),
                      ),
                      child: Column(
                        mainAxisSize: MainAxisSize.min,
                        children: [
                          if (pickerType == PickerType.JHijri)
                            JCalendarDatePicker(
                              initialDate: jCheckDate(selectedDate),
                              localeCode: locale?.languageCode ?? "en",
                              firstDate: jCheckDate(startDate),
                              initialCalendarMode:
                                  pickerMode ?? DatePickerMode.day,
                              lastDate: jCheckDate(endDate),
                              selectableDayPredicate: jSelectableDayPredicate,
                              onDateChanged: onChange != null
                                  ? (dateTime) => onChange!(JPickerValue.value(
                                      JDateModel(
                                          jhijri: JHijri(
                                              fDate: dateTime.dateTime))))
                                  : (dateTime) => selected = JPickerValue.value(
                                      JDateModel(
                                          jhijri: JHijri(
                                              fDate: dateTime.dateTime))),
                            ),
                          if (pickerType == PickerType.JNormal)
                            CalendarDatePicker(
                              initialDate: checkDate(selectedDate),
                              firstDate: checkDate(startDate),
                              lastDate: checkDate(endDate),
                              initialCalendarMode:
                                  pickerMode ?? DatePickerMode.day,
                              selectableDayPredicate: selectableDayPredicate,
                              onDateChanged: onChange != null
                                  ? (dateTime) => onChange!(JPickerValue.value(
                                      JDateModel(dateTime: dateTime)))
                                  : (dateTime) => selected = JPickerValue.value(
                                      JDateModel(dateTime: dateTime)),
                            ),
                        ],
                      ),
                    ),
                    if (buttons != null) buttons!,
                    if (buttons == null)
                      Container(
                        decoration: BoxDecoration(
                          color: backgroundColor ?? Colors.white,
                          borderRadius: BorderRadius.only(
                            bottomLeft:
                                borderRadius ?? const Radius.circular(16),
                            bottomRight:
                                borderRadius ?? const Radius.circular(16),
                          ),
                        ),
                        child: Row(
                          mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                          mainAxisSize: MainAxisSize.max,
                          children: [
                            Expanded(
                              child: TextButton(
                                style: ButtonStyle(
                                  backgroundColor: MaterialStateProperty.all(
                                      cancelButtonColor),
                                ),
                                onPressed: onCancel ??
                                    () {
                                      Navigator.pop(context);
                                    },
                                child: Text(
                                  cancelButtonText ?? "Cancel",
                                  style: TextStyle(
                                      color: buttonTextColor ?? Colors.black),
                                ),
                              ),
                            ),
                            const SizedBox(
                              height: 20,
                              child: VerticalDivider(
                                color: Colors.grey,
                              ),
                            ),
                            Expanded(
                              child: TextButton(
                                style: ButtonStyle(
                                  backgroundColor:
                                      MaterialStateProperty.all(okButtonColor),
                                ),
                                onPressed: onOk != null
                                    ? selected != null
                                        ? () => onOk!(selected!)
                                        : () => onOk!(JPickerValue.value(
                                            JDateModel(
                                                dateTime: DateTime.now())))
                                    : () {
                                        Navigator.pop(context, selected);
                                      },
                                child: Text(
                                  okButtonText ?? "Ok",
                                  style: TextStyle(
                                      color: buttonTextColor ?? Colors.black),
                                ),
                              ),
                            ),
                          ],
                        ),
                      ),
                  ],
                ),
              ),
            )));
  }

  DateTime checkDate(JDateModel? da) {
    DateTime da2 = DateTime.now();
    if (da != null) {
      if (da.jhijri != null) {
        da2 = da.jhijri!.dateTime;
      } else if (da.dateTime != null) {
        da2 = da.dateTime!;
      }
    }
    return da2;
  }

  JHijri jCheckDate(JDateModel? da) {
    JHijri da2 = JHijri.now();
    if (da != null) {
      if (da.jhijri != null) {
        da2 = da.jhijri!;
      } else if (da.dateTime != null) {
        da2 = JHijri(fDate: da.dateTime);
      }
    }
    return da2;
  }
}


lib/_src/_jDateTimePicker.dart
import 'package:flutter/material.dart';
import '../jhijri_picker.dart';

Future<DateTime?> showJDateTimePicker({
  required BuildContext context,
  JDateModel? selectedDate,
  JDateModel? startDate,
  JDateModel? endDate,
  DatePickerMode pickerMode = DatePickerMode.day,
  String? okButton,
  String? cancelButton,
  Widget? headerTitle,
  Widget? buttons,
  Function(DateTime datetime)? onOk,
  Function(DateTime datetime)? onChange,
  VoidCallback? onCancel,
  ThemeData? theme,
  Color? primaryColor,
  Color? backgroundColor,
  Color? calendarTextColor,
  Color? buttonTextColor,
  Radius? borderRadius,
  TextDirection? textDirection,
  Locale? locale,
}) {
  return showDialog<DateTime?>(
    context: context,
    builder: (BuildContext context) {
      return JDateTimePicker(
        startDate: startDate,
        selectedDate: selectedDate,
        endDate: endDate,
        pickerMode: pickerMode,
        themeD: theme,
        textDirection: textDirection,
        locale: locale,
        okButtonT: okButton,
        cancelButtonT: cancelButton,
        onOk: onOk,
        onCancel: onCancel,
        primaryColor: primaryColor,
        calendarTextColor: calendarTextColor,
        backgroundColor: backgroundColor,
        borderRadius: borderRadius,
        buttonTextColor: buttonTextColor,
        headerTitle: headerTitle,
        buttons: buttons,
        widgetType: WidgetType.JDialog,
        onChange: onChange,
      );
    },
  );
}

class JDateTimePicker extends StatefulWidget {
  final JDateModel? selectedDate;
  final JDateModel? startDate;
  final JDateModel? endDate;
  final DatePickerMode? pickerMode;
  final String? okButtonT;
  final String? cancelButtonT;
  final Widget? headerTitle;
  final Function(DateTime datetime)? onOk;
  final Function(DateTime datetime)? onChange;
  final VoidCallback? onCancel;
  final ThemeData? themeD;
  final Color? primaryColor;
  final Color? backgroundColor;
  final Color? calendarTextColor;
  final Color? buttonTextColor;
  final Radius? borderRadius;
  final Widget? buttons;
  final WidgetType widgetType;
  final TextDirection? textDirection;
  final Locale? locale;
  const JDateTimePicker(
      {Key? key,
      this.selectedDate,
      this.startDate,
      this.endDate,
      this.pickerMode,
      this.okButtonT,
      this.cancelButtonT,
      this.headerTitle,
      this.onOk,
      this.onChange,
      this.onCancel,
      this.themeD,
      this.primaryColor,
      this.backgroundColor,
      this.calendarTextColor,
      this.buttonTextColor,
      this.borderRadius,
      this.buttons,
      this.widgetType = WidgetType.JDialog,
      this.textDirection,
      this.locale})
      : super(key: key);

  @override
  State<JDateTimePicker> createState() => _JDateTimePickerState();
}

class _JDateTimePickerState extends State<JDateTimePicker> {
  late DateTime sel;
  late DateTime sel1;
  late DateTime sta;
  late DateTime end;
  @override
  void initState() {
    _jDa();
    super.initState();
  }

  _jDa() {
    if (widget.startDate != null) {
      if (widget.startDate!.jhijri != null) {
        sta = widget.startDate!.jhijri!.dateTime;
      } else if (widget.startDate!.dateTime != null) {
        sta = widget.startDate!.dateTime!;
      } else {
        sta = DateTime(2018);
      }
    }
    if (widget.endDate != null) {
      if (widget.endDate!.jhijri != null) {
        end = widget.endDate!.jhijri!.dateTime;
      } else if (widget.endDate!.dateTime != null) {
        end = widget.endDate!.dateTime!;
      } else {
        end = DateTime(2060);
      }
    }
    if (widget.selectedDate != null) {
      if (widget.selectedDate!.jhijri != null) {
        sel1 = widget.selectedDate!.jhijri!.dateTime;
        sel = widget.selectedDate!.jhijri!.dateTime;
      } else if (widget.selectedDate!.dateTime != null) {
        sel1 = widget.selectedDate!.dateTime!;
        sel = widget.selectedDate!.dateTime!;
      } else {
        sel1 = DateTime.now();
        sel = DateTime.now();
      }
    }
  }

  @override
  Widget build(BuildContext context) {
    final theme = widget.themeD ?? Theme.of(context);
    final media = MediaQuery.of(context);
    return _JLocalDirection(
        locale: widget.locale,
        textDirection: widget.textDirection,
        child: _WidgetType(
            widgetType: widget.widgetType,
            child: Theme(
              data: ThemeData(
                colorScheme: theme.colorScheme.copyWith(
                  primary: widget.primaryColor ?? Colors.blue,
                  surface: widget.backgroundColor ?? Colors.white,
                  onSurface: widget.calendarTextColor ?? Colors.black,
                ),
              ),
              child: SingleChildScrollView(
                child: Column(
                  mainAxisSize: MainAxisSize.min,
                  children: [
                    if (widget.headerTitle != null) widget.headerTitle!,
                    Container(
                      constraints:
                          BoxConstraints(maxHeight: media.size.height - 120),
                      decoration: BoxDecoration(
                        color: widget.backgroundColor ?? Colors.white,
                        borderRadius: BorderRadius.only(
                            topLeft: widget.borderRadius ??
                                const Radius.circular(16),
                            topRight: widget.borderRadius ??
                                const Radius.circular(16)),
                      ),
                      child: Column(
                        mainAxisSize: MainAxisSize.min,
                        children: [
                          CalendarDatePicker(
                            initialDate: sel1,
                            firstDate: sta,
                            lastDate: end,
                            onDateChanged: widget.onChange != null
                                ? (dateTime) => widget.onChange!(dateTime)
                                : (dateTime) {
                                    sel = dateTime;
                                  },
                          ),
                        ],
                      ),
                    ),
                    if (widget.buttons != null) widget.buttons!,
                    if (widget.buttons == null)
                      Container(
                        decoration: BoxDecoration(
                          color: widget.backgroundColor ?? Colors.white,
                          borderRadius: BorderRadius.only(
                            bottomLeft: widget.borderRadius ??
                                const Radius.circular(16),
                            bottomRight: widget.borderRadius ??
                                const Radius.circular(16),
                          ),
                        ),
                        child: Row(
                          mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                          mainAxisSize: MainAxisSize.max,
                          children: [
                            Expanded(
                              child: TextButton(
                                style: ButtonStyle(
                                  backgroundColor: MaterialStateProperty.all(
                                      widget.backgroundColor),
                                ),
                                onPressed: widget.onCancel ??
                                    () {
                                      Navigator.pop(context);
                                    },
                                child: Text(
                                  widget.cancelButtonT ?? "Cancel",
                                  style: TextStyle(
                                      color: widget.buttonTextColor ??
                                          Colors.black),
                                ),
                              ),
                            ),
                            const SizedBox(
                              height: 20,
                              child: VerticalDivider(
                                color: Colors.grey,
                              ),
                            ),
                            Expanded(
                              child: TextButton(
                                style: ButtonStyle(
                                  backgroundColor: MaterialStateProperty.all(
                                      widget.backgroundColor),
                                ),
                                onPressed: widget.onOk != null
                                    ? () => widget.onOk!(sel)
                                    : () {
                                        Navigator.pop(context, sel);
                                      },
                                child: Text(
                                  widget.okButtonT ?? "Ok",
                                  style: TextStyle(
                                      color: widget.buttonTextColor ??
                                          Colors.black),
                                ),
                              ),
                            ),
                          ],
                        ),
                      ),
                  ],
                ),
              ),
            )));
  }
}

class _JLocalDirection extends StatelessWidget {
  final TextDirection? textDirection;
  final Locale? locale;
  final Widget child;
  const _JLocalDirection(
      {Key? key, this.textDirection, this.locale, required this.child})
      : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Localizations.override(
      context: context,
      locale: locale,
      child: textDirection != null
          ? Directionality(
              textDirection: textDirection!,
              child: child,
            )
          : child,
    );
  }
}

class _WidgetType extends StatelessWidget {
  final Widget child;
  final WidgetType widgetType;
  const _WidgetType({Key? key, required this.widgetType, required this.child})
      : super(key: key);

  @override
  Widget build(BuildContext context) {
    switch (widgetType) {
      case WidgetType.JDialog:
        return Dialog(
          backgroundColor: Colors.transparent,
          alignment: Alignment.center,
          child: child,
        );
      case WidgetType.JContainer:
        return Container(
          color: Colors.transparent,
          alignment: Alignment.center,
          child: child,
        );
    }
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:jhijri/jHijri.dart';
import 'package:jhijri_picker/jhijri_picker.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'JHijriPicker',
      theme: ThemeData(
        primarySwatch: Colors.deepPurple,
      ),
      debugShowCheckedModeBanner: false,
      home: const HomePage(),
    );
  }
}

class HomePage extends StatelessWidget {
  const HomePage({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("JHijriDatePicker"),
        actions: [
          Padding(
            padding: const EdgeInsets.all(8.0),
            child: FloatingActionButton.extended(
              heroTag: "4",
              onPressed: () async {
                final dateTime = await showGlobalDatePicker(
                    context: context, pickerType: PickerType.JNormal);
                if (dateTime != null) {
                  debugPrint(dateTime.toString());
                }
              },
              tooltip: 'Normal Date',
              icon: const Icon(Icons.date_range_outlined),
              label: Text("Normal"),
            ),
          ),
          Padding(
            padding: const EdgeInsets.all(8.0),
            child: FloatingActionButton.extended(
              heroTag: "3",
              onPressed: () async {
                final dateTime = await showGlobalDatePicker(
                  context: context,
                  startDate: JDateModel(
                      jhijri: JHijri(
                    fYear: 1442,
                    fMonth: 12,
                    fDay: 10,
                  )),
                  selectedDate: JDateModel(jhijri: JHijri.now()),
                  endDate: JDateModel(
                      jhijri: JHijri(
                    fDay: 25,
                    fMonth: 1,
                    fYear: 1460,
                  )),
                  pickerMode: DatePickerMode.day,
                  pickerTheme: Theme.of(context),
                  textDirection: TextDirection.rtl,
                  //locale: const Locale("ar", "SA"),
                  okButtonText: "حفظ",
                  cancelButtonText: "عودة",
                  onOk: (value) {
                    debugPrint(value.toString());
                    Navigator.pop(context);
                  },
                  onCancel: () {
                    Navigator.pop(context);
                  },
                  primaryColor: Colors.blue,
                  calendarTextColor: Colors.white,
                  backgroundColor: Colors.black,
                  borderRadius: const Radius.circular(10),
                  buttonTextColor: Colors.white,
                  headerTitle: const Center(
                    child: Text(
                      "التقويم الهجري",
                      style: TextStyle(color: Colors.white),
                    ),
                  ),
                );
                if (dateTime != null) {
                  debugPrint(dateTime.toString());
                }
              },
              tooltip: 'Hijri Date',
              icon: const Icon(Icons.date_range_outlined),
              label: Text("Hijri"),
            ),
          ),
        ],
      ),
      body: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: <Widget>[
          Expanded(
              child: ListView(
            children: [
              JGlobalDatePicker(
                widgetType: WidgetType.JContainer,
                pickerType: PickerType.JHijri,
                buttons: const SizedBox(),
                primaryColor: Colors.blue,
                calendarTextColor: Colors.white,
                backgroundColor: Colors.black,
                borderRadius: const Radius.circular(10),
                headerTitle: const Center(
                  child: Text("التقويم الهجري"),
                ),
                startDate: JDateModel(dateTime: DateTime.parse("1984-12-24")),
                selectedDate: JDateModel(dateTime: DateTime.now()),
                endDate: JDateModel(dateTime: DateTime.parse("2030-09-20")),
                pickerMode: DatePickerMode.day,
                pickerTheme: Theme.of(context),
                textDirection: TextDirection.rtl,
                onChange: (val) {
                  debugPrint(val.toString());
                },
              ),
              const Divider(
                color: Colors.blue,
              ),
              JGlobalDatePicker(
                widgetType: WidgetType.JContainer,
                pickerType: PickerType.JNormal,
                buttons: const SizedBox(),
                primaryColor: Colors.blue,
                calendarTextColor: Colors.white,
                backgroundColor: Colors.black,
                borderRadius: const Radius.circular(10),
                headerTitle: const Center(
                  child: Text("التقويم الميلادي"),
                ),
                startDate: JDateModel(dateTime: DateTime.parse("1984-12-24")),
                selectedDate: JDateModel(dateTime: DateTime.now()),
                endDate: JDateModel(dateTime: DateTime.parse("2030-09-20")),
                pickerMode: DatePickerMode.day,
                pickerTheme: Theme.of(context),
                textDirection: TextDirection.rtl,
                onChange: (val) {
                  debugPrint(val.toString());
                },
              ),
            ],
          ))
        ],
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}
